numba/tests/test_array_reductions.py

```diff
@@ -938,6 +938,57 @@ def a_variations():
             for a in a_variations():
                 check(a)
 
+    def test_argmax_axis_1d_2d_3d(self):
+        arr1d = np.array([0, 20, 3, 4])
+        arr2d = np.arange(6).reshape(2, 3)
+        arr2d[0,1] += 100
+
+        arr3d = np.arange(24).reshape(2, 3, 4) + 10
+        arr3d[0,1,1] += 100
+        arr3d[1,0,0] += 100
+
+        for arr in [arr1d, arr2d, arr3d]:
+            axises = list(range(arr.ndim)) + [
+                -(i+1) for i in range(arr.ndim)
+            ]
+            py_functions = [
+                lambda a, _axis=axis: np.argmax(a, axis=_axis)
+                for axis in axises
+            ]
+            c_functions = [
+                jit(nopython=True)(pyfunc) for pyfunc in py_functions
+            ]
+            for (pyfunc, cfunc) in zip(py_functions, c_functions):
+                self.assertPreciseEqual(pyfunc(arr), cfunc(arr))
+
+    def test_argmax_axis_out_of_range(self):
+        arr1d = np.arange(6)
+        arr2d = np.arange(6).reshape(2, 3)
+
+        @jit(nopython=True)
+        def jitargmax(arr, axis):
+            return np.argmax(arr, axis)
+
+        def assert_raises(arr, axis):
+            with self.assertRaises(ValueError):
+                np.argmax(arr, axis)
+            with self.assertRaises(ValueError):
+                jitargmax(arr, axis)
```

Comment:
The Numba code base tends to use this pattern to ensure that exceptions are being raised from the expected places.
https://github.com/numba/numba/blob/e314821f48bfc1678c9662584eef166fb9d5469c/numba/tests/test_array_reductions.py#L477-L482